sound/soc/sof/sof-audio.c

```diff
@@ -189,12 +190,25 @@ int sof_restore_pipelines(struct device *dev)
 		switch (swidget->id) {
 		case snd_soc_dapm_dai_in:
 		case snd_soc_dapm_dai_out:
+			ipc_size = sizeof(struct sof_ipc_comp_dai) +
+				   sizeof(struct sof_ipc_comp_new_ext);
+			comp = kzalloc(ipc_size, GFP_KERNEL);
+			if (!comp)
+				return -ENOMEM;
+
 			dai = swidget->private;
-			comp_dai = &dai->comp_dai;
-			ret = sof_ipc_tx_message(sdev->ipc,
-						 comp_dai->comp.hdr.cmd,
-						 comp_dai, sizeof(*comp_dai),
+			/* restore the struct sof_ipc_comp_dai */
```

Comment:
it's not really 'restore', but 'copy base information for sof_ipc_comp_dai'

Reply:
maybe 'retrieve'?

Reply:
"copy base data first"

Or remove this comment altogether, you already have an 'append' explanation below

Reply:
removed now.